server/base/src/main/java/org/apache/accumulo/server/fs/VolumeManagerImpl.java

```diff
@@ -384,20 +390,36 @@ public short getDefaultReplication(Path path) {
    * @param filesystemURI Volume Filesystem URI
    * @return Hadoop Configuration with custom overrides for this FileSystem
    */
-  private static Configuration getVolumeManagerConfiguration(AccumuloConfiguration conf,
+  protected static Configuration getVolumeManagerConfiguration(AccumuloConfiguration conf,
       final Configuration hadoopConf, final String filesystemURI) {
 
-    final Configuration volumeConfig = new Configuration(hadoopConf);
+    Map<String,String> volumeHdfsConfigOverrides =
```

Comment:
Is it ok to do the following and void the computation of volumeHdfsConfigOverrides on each call?

```java
    final String key = hadoopConf.hashCode() + filesystemURI;
    return HDFS_CONFIGS_FOR_VOLUME.get(key, (fs) -> {

      Map<String,String> volumeHdfsConfigOverrides =
              conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX).entrySet()
                      .stream().filter(e -> e.getKey().startsWith(filesystemURI + ".")).collect(
                              Collectors.toUnmodifiableMap(e -> e.getKey().substring(filesystemURI.length() + 1),
                                      Entry::getValue));
      if (volumeHdfsConfigOverrides.isEmpty()) {
        return hadoopConf;
      }else {
        log.debug("Caching a new configuration for volume: {} with overrides: {}", filesystemURI,
                volumeHdfsConfigOverrides);
        Configuration volumeConfig = new Configuration(hadoopConf);
        volumeHdfsConfigOverrides.forEach((k, v) -> {
          log.info("Overriding property {}={} for volume {}", k, v, filesystemURI);
          volumeConfig.set(k, v);
        });
        return new Configuration(volumeConfig);
      }
    });
```

Reply:
My only concern with putting more into the `Cache.get` call is this in the javadoc:
```
   * Some attempted update operations on this cache by other threads may be blocked while the
   * computation is in progress, so the computation should be short and simple, and must not attempt
   * to update any other mappings of this cache.
```

Reply:
Could consider this change later if the computation of volumeHdfsConfigOverrides actually shows up in any profiling.

Reply:
I included this suggestion in my code review commit 8f1ee93431999e673a878be134f947270eb0dd17